google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryImpl.java

```diff
@@ -265,13 +274,15 @@ public boolean delete(String datasetId, DatasetDeleteOption... options) {
   }
 
   @Override
-  public boolean delete(final DatasetId datasetId, DatasetDeleteOption... options) {
+  public boolean delete(DatasetId datasetId, DatasetDeleteOption... options) {
+    final DatasetId completeDatasetId = datasetId.setProjectId(options().projectId());
```

Comment:
`setProjectId` does not override the value if it's already set, it just set if the user did not provide one (is `null`).